app/src/main/java/me/saket/dank/data/DankSqliteOpenHelper.kt

```diff
@@ -17,6 +18,7 @@ class DankSqliteOpenHelper(context: Context) : SQLiteOpenHelper(context, DB_NAME
     db.execSQL(CachedMessage.QUERY_CREATE_TABLE)
     db.execSQL(PendingSyncReply.QUERY_CREATE_TABLE)
     db.execSQL(AppShortcut.QUERY_CREATE_TABLE)
+    db.execSQL(AccountManager.QUERY_CREATE_TABLE)
   }
 
   override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
```

Comment:
The upgrade method will need some fixing. I can see that you followed the convention in the first if statement but it won't work correctly.

Let's say that the user is on the version of app where the database has version 1. Now when they update to latest version this method will be called with parameters `oldVersion=1` and `newVersion=3`. In that case by following the logic inside of if statements we can see that none of them will be executed and the upgrade won't work correctly.

In the first statement you want to keep on check on `oldVersion` only. That is, we should simply delete cached messages table when upgrading from that version.

In the second statement you want to check only on `newVersion` since you need to create the table also when upgrading from older versions. Additionally I'm not sure but perhaps we should go a bit further and check whether new version is equal to or greater than 3 to also take into account future versions.

___

Other than that it would be a good idea to add tests for the whole upgrade since that is an important operation. Of course tests for other operations on the database would be nice too but I feel like this one is the minimum.